code/datums/flock/flock.dm

```diff
@@ -363,49 +637,84 @@
 
 /datum/flock/proc/process()
 	var/list/floors_no_longer_existing = list()
-	// check all active floors
+
 	for(var/turf/simulated/floor/feather/T in src.all_owned_tiles)
 		if(!T || T.loc == null || T.broken)
-			// tile got killed, remove it
 			floors_no_longer_existing |= T
 			continue
-		// check adjacent tiles to see if we've been surrounded and can start generating, or if we're no longer surrounded and can't generate
-		// var/validNeighbors = 0
-		// var/list/neighbors = getNeighbors(T, cardinal)
-		// for(var/turf/simulated/floor/feather/F in neighbors)
-		// 	validNeighbors += 1
-		// if(validNeighbors < 4 && T.generating)
-		// 	T.off()
-		// else if(validNeighbors >= 4 && !T.generating)
-		// 	T.on()
-
-	if(floors_no_longer_existing.len > 0)
+
+	if(length(floors_no_longer_existing))
 		src.all_owned_tiles -= floors_no_longer_existing
 
+	for(var/datum/unlockable_flock_structure/ufs as anything in src.unlockableStructures)
+		ufs.process()
+
+	var/turf/busy_turf
+	for(var/name in src.busy_tiles)
+		busy_turf = src.busy_tiles[name]
+		if (QDELETED(busy_turf))
+			src.unreserveTurf(busy_turf)
+
+	for(var/turf/T in src.priority_tiles)
+		if (QDELETED(T))
+			src.togglePriorityTurf(T)
+
+	for(var/atom/S in src.deconstruct_targets)
+		if(QDELETED(S))
+			src.toggleDeconstructionFlag(S)
+
+	var/atom/M
+	for(var/enemy in src.enemies)
+		M = src.enemies[enemy]["mob"]
+		if (QDELETED(M))
+			src.removeEnemy(M)
+
 /datum/flock/proc/convert_turf(var/turf/T, var/converterName)
 	src.unreserveTurf(converterName)
 	src.claimTurf(flock_convert_turf(T))
-	playsound(T, "sound/items/Deconstruct.ogg", 70, 1)
-
+	playsound(T, "sound/items/Deconstruct.ogg", 40, 1)
+
+///Unlock an achievement (string) if it isn't already unlocked
+/datum/flock/proc/achieve(var/str)
+	src.achievements |= str
+	var/datum/abilityHolder/flockmind/aH = src.flockmind.abilityHolder
+	aH?.updateCompute()
+
+/datum/flock/proc/unAchieve(var/str)
+	src.achievements -= str
+	var/datum/abilityHolder/flockmind/aH = src.flockmind.abilityHolder
+	aH?.updateCompute()
+
+///Unlock an achievement (string) if it isn't already unlocked
+/datum/flock/proc/hasAchieved(var/str)
+	return (str in src.achievements)
 ////////////////////
 // GLOBAL PROCS!!
 ////////////////////
 
 // made into a global proc so a reagent can use it
 // simple enough: if object path matches key, replace with instance of value
 // if value is null, just delete object
+// !!!! priority is determined by list order !!!!
+// if you have a subclass, it MUST go first in the list, or the first type that matches will take priority (ie, the superclass)
+// see /obj/machinery/light/small/floor and /obj/machinery/light for examples of this
 /var/list/flock_conversion_paths = list(
-	/obj/grille = /obj/grille/flock,
-	/obj/window = /obj/window/feather,
+	/obj/grille/steel = /obj/grille/flock,
```

Comment:
The idea was that catwalks would be converted directly into flock floor instead of lattices since they about the same, which is coded below, but can be changed too

Reply:
catwalk-to-turf seems fine, but I still don't like that only steel grilles can be converted into flockgrilles